bridges/snowbridge/pallets/inbound-queue/src/lib.rs

```diff
@@ -76,6 +76,42 @@ pub use pallet::*;
 
 pub const LOG_TARGET: &str = "snowbridge-inbound-queue";
 
+pub trait RewardProcessor<T: frame_system::Config> {
+	fn process_reward(who: T::AccountId, message: Message) -> DispatchResult;
+}
+
+impl<T: frame_system::Config> RewardProcessor<T> for () {
+	fn process_reward(_who: T::AccountId, _message: Message) -> DispatchResult {
+		Ok(())
+	}
+}
+
+pub struct DeliveryCostReward<T>(sp_std::marker::PhantomData<T>);
```

Comment:
I would name this `RewardThroughSovereign` or something like that. Otherwise looks good